commands.js

```diff
@@ -286,6 +286,46 @@ var commands = exports.commands = {
 	},
 	makechatroomhelp: ["/makechatroom [roomname] - Creates a new room named [roomname]. Requires: ~"],
 
+	personalroom: function (target, room, user) {
+		// First and foremost, check the number of personal rooms owned by the user.
+		if (user.personalRooms && user.personalRooms > 4) return this.sendReply("You can't own more than 5 personal rooms.");
+
+		// Chosen name have the same limitations as regular rooms.
+		if (target.includes(',') || target.includes('|') || target.includes('[') || target.includes('-')) {
+			return this.sendReply("Room titles can't contain any of: ,|[-");
+		}
+
+		// Check if the custom part of the room title is empty. It needs a title.
+		var id = toId(target);
+		if (!id) return this.parse('/help personalroom');
+
+		// Create the name of the room using user's ID and the chosen name.
+		var roomName = target + " (" + user.userid + "'s room)";
+		id = toId(roomName);
+
+		// Personal rooms can always be created, but the name must still be unique.
+		if (Rooms.search(id)) return this.sendReply("You already have created room '" + roomName + "'.");
```

Comment:
Just say it already exists. This prevents strange shenanigans if a server admin creates a room called "This is Joim's room" and then Joim tries to make a personal room called "This is".